pytest-embedded-idf/tests/test_idf.py

```diff
@@ -987,3 +987,54 @@ def test_python_case(dut):
     assert junit_report[0].attrib['is_unity_case'] == '0'  # Python test case
     for testcase in junit_report[1:]:
         assert testcase.attrib['is_unity_case'] == '1'  # Other test cases
+
+
+def test_skip_if_soc(testdir):
+    EMBEDDED_SERVICES = ['--embedded-services', 'esp,idf']
+
+    def run_pytest_with_target(target):
+        count = len(target.split('|'))
+        return testdir.runpytest(*EMBEDDED_SERVICES, '--target', target, '--count', count)
+
+    testdir.makepyfile("""
+        import pytest
+        from idf_build_apps.constants import SUPPORTED_TARGETS
+
+        @pytest.mark.skip_if_soc("SOC_ULP_LP_UART_SUPPORTED == 1")
```

Comment:
Are more complex conditions supported, or do we have the same limitation as in idf-build-apps, where we can't write `A == 1 or B == 2 or C == 3`?

Reply:
good point. I'll fix this limitation together while moving the if parser to a separate package.